packages/x-tree-view/src/internals/useTreeView/useTreeView.ts

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import useForkRef from '@mui/utils/useForkRef';
+import { EventHandlers } from '@mui/base/utils';
+import {
+  TreeViewAnyPluginSignature,
+  TreeViewInstance,
+  TreeViewPlugin,
+  ConvertPluginsIntoSignatures,
+  MergePluginsProperty,
+} from '../models';
+import {
+  UseTreeViewDefaultizedParameters,
+  UseTreeViewParameters,
+  UseTreeViewReturnValue,
+  UseTreeViewRootSlotProps,
+} from './useTreeView.types';
+import { DEFAULT_TREE_VIEW_CONTEXT_VALUE } from '../TreeViewProvider/TreeViewContext';
+import { useTreeViewModels } from './useTreeViewModels';
+import { TreeViewContextValue } from '../TreeViewProvider';
+import { TREE_VIEW_CORE_PLUGINS } from '../corePlugins';
+
+export const useTreeView = <Plugins extends readonly TreeViewPlugin<TreeViewAnyPluginSignature>[]>(
+  inParams: UseTreeViewParameters<Plugins>,
+): UseTreeViewReturnValue<ConvertPluginsIntoSignatures<Plugins>> => {
+  const plugins = [...TREE_VIEW_CORE_PLUGINS, ...inParams.plugins];
+  type Signatures = ConvertPluginsIntoSignatures<typeof plugins>;
+
+  const params = plugins.reduce((acc, plugin) => {
+    if (plugin.getDefaultizedParams) {
+      return plugin.getDefaultizedParams(acc);
+    }
+
+    return acc;
+  }, inParams) as unknown as UseTreeViewDefaultizedParameters<Plugins>;
+
+  const models = useTreeViewModels(
+    plugins,
+    params as MergePluginsProperty<Signatures, 'defaultizedParams'>,
+  );
+  const instanceRef = React.useRef<TreeViewInstance<Signatures>>(
+    {} as TreeViewInstance<Signatures>,
+  );
+  const instance = instanceRef.current;
+  const innerRootRef = React.useRef(null);
+  const handleRootRef = useForkRef(innerRootRef, inParams.rootRef);
+
+  const [state, setState] = React.useState(() => {
+    const temp = {} as MergePluginsProperty<Signatures, 'state'>;
+    plugins.forEach((plugin) => {
+      if (plugin.getInitialState) {
+        Object.assign(
+          temp,
+          plugin.getInitialState(params as UseTreeViewDefaultizedParameters<any>),
+        );
+      }
+    });
+
+    return temp;
+  });
+
+  const rootPropsGetters: (<TOther extends EventHandlers = {}>(
+    otherHandlers: TOther,
+  ) => React.HTMLAttributes<HTMLUListElement>)[] = [];
+  let contextValue: TreeViewContextValue<Signatures> = DEFAULT_TREE_VIEW_CONTEXT_VALUE;
+
+  const runPlugin = (plugin: TreeViewPlugin<any>) => {
+    const pluginResponse =
+      plugin({ instance, params, state, setState, rootRef: innerRootRef, models }) || {};
+
+    if (pluginResponse.getRootProps) {
+      rootPropsGetters.push(pluginResponse.getRootProps);
+    }
+
+    if (pluginResponse.contextValue) {
+      contextValue = pluginResponse.contextValue;
+    }
+  };
+
+  plugins.forEach(runPlugin);
+
+  const getRootProps = <TOther extends EventHandlers = {}>(
```

Comment:
FYI, we're in process of changing Base UI to accept all props here: https://github.com/mui/material-ui/issues/38186.
It won't change the logic of this particular hook, just the naming and type.

Reply:
On this PR, would it just mean renaming `otherHandlers into `otherProps`?